samples/client/petstore/python-experimental/petstore_api/model_utils.py

```diff
@@ -848,6 +848,10 @@ def deserialize_model(model_data, model_class, path_to_item, check_type,
         used_model_class = model_class.get_discriminator_class(
             from_server, model_data)
 
+    # This is the case if the used_model_class isn't a subclass but rather the parent class itself
+    if not used_model_class:
```

Comment:
Our old code always returned a model class.
Why do we need to return None now?
Can you remove the case where this is returning None and just returns the correct class?

Reply:
This was actually the original error I was looking at before I started looking at the recursive discriminator issue. The method returns nothing when the discriminator specified is not in the discriminator mapping. I believe in that case, a reasonable solution is to return the base class itself, but perhaps we should instead return an error here?

Reply:
Per the spec here: https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md#composition-and-inheritance-polymorphism
If a discriminator is used, then the child class must contain the discriminator propertyName, and by inference the child class must be in the discriminator map.
Please raise an exception that lets the user know what went wrong.